packages/@ember/application/lib/application.js

```diff
@@ -387,6 +387,16 @@ const Application = Engine.extend({
     @return {ApplicationInstance} the application instance
   */
   buildInstance(options = {}) {
+    assert(
+      'You cannot build new instances of this application since it is being destroyed',
+      this.isDestroying
+    );
+
+    assert(
+      'You cannot build new instances of this application since it has already been destroyed',
+      this.isDestroyed
```

Comment:
This assertion needs to be moved up above the other one. If we don't, it can never be triggered (since we will never have `this.isDestroying` be false and `this.isDestroyed` be true).

Also, shouldn't the assertion be `!this.isDestroyed` here?

Reply:
🤦